
(base) (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda deactivate
(.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda env create -f my_environment.yml
Retrieving notices: ...working... done

CondaValueError: prefix already exists: C:\ProgramData\anaconda3\envs\mlops-env

(.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda env list
# conda environments:
#
base                     C:\ProgramData\anaconda3
mlops-env                C:\ProgramData\anaconda3\envs\mlops-env

(.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3>

(.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda remove --name mlops-env  --all

 (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda env create -f my_environment.yml

 (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda activate mlops-env
 (mlops-env) (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3>

(mlops-env) (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> python --version
Python 3.12.11

#Tem que ser via linha de comando pois o intelliJ community não tem a integração
(mlops-env) (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> python Main.py

#Comando pytest vai buscar por arquivo começado (test_) ou terminado (_test_ com test
(mlops-env) (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> pytest

#Comando para rodar um script .py
(mlops-env) (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> python Main.py

#Atualizando versão do pacote para o Jupyter (Com VS Code) funcionar no ambiente do mlops-env
(mlops-env) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda install -n mlops-env ipykernel

#instalando pacote do arquivo que está no .ipynb
(mlops-env) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda install -c conda-forge tensorflow


#Comando para mlflow visual setando onde os modelos estão sendo criados (mlflow.set_tracking_uri("file:///C:/Users/Raphael/IdeaProjects/MLUnidade3/mlruns"))
mlflow ui --backend-store-uri "file:///C:/Users/Raphael/IdeaProjects/MLUnidade3/mlruns"

#Deu erro pois haviam 2 ambientes rodando!!!
(mlops-env) (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> mlflow ui
ImportError: DLL load failed while importing _ctypes: The specified module could not be found.

(mlops-env) (.venv) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> deactivate
(mlops-env) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> mlflow ui
INFO:waitress:Serving on http://127.0.0.1:5000


#problema do ctypes
(mlops-env) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda deactivate
(base) PS C:\Users\Raphael\IdeaProjects\MLUnidade3> conda update conda
Collecting package metadata (current_repodata.json): done
Solving environment: done




